backend/logic/courses/courses.js

```diff
@@ -1,19 +1,72 @@
-const mysql = require('../../sql/connection');
+const mysql = require("../../sql/connection");
+const format = require("../../validation/format")
 
-async function queryCourseByTag(tag) {
-    if (!tag) {
-        throw Error("Undefined tag")
-    }
+var queryCourseByTag = async function queryCourseByTag(tag) {
+  if (!tag) {
+    throw Error("Undefined tag")
+  }
+
+  let connection = await mysql.getNewConnection();
+  try {
+    let courses = await connection.query("SELECT course_code FROM course_tag WHERE course_tag.tag_name LIKE ?", tag);
+    connection.release();
+    return (JSON.parse(JSON.stringify(courses)));
+  } catch (error) {
+    connection.release();
+    console.error(error);
+    throw Error(error.message);
+  }
+}
+
+var addCompletedCourses = async (studentId, courses, section) => {
+  if (!format.verifyStudentId(studentId)) {
+    throw new Error("invalid format id");
+  }
+  await format.verifyCompletedCourses(courses);
+
+  let connection;
+  try {
+    connection = await mysql.getNewConnection();
+  } catch (error) {
+    throw new Error("failed to establish connection with database");
+  }
 
-    let connection = await mysql.getNewConnection();
-    try {
-        let courses = await connection.query('SELECT course_code FROM course_tag WHERE course_tag.tag_name LIKE ?', tag);
-        connection.release();
-        return (JSON.parse(JSON.stringify(courses)));
-    } catch (error) {
-        connection.release();
-        console.error(error);
-        throw Error(error.message);
+  try {
+    await connection.beginTransaction();
+    // Obtain all current course offerings and put them into a hashtable for fast lookup
+    let results = await connection.query("SELECT * FROM course_offerings;");
```

Comment:
I think you can just select the id, semester and section from this table instead of all columns

Reply:
changed thanks!